themes/default/content/blog/mlops-huggingface-llm-aws-sagemaker-python/index.md

```diff
@@ -0,0 +1,131 @@
+
+## Instructions
+
+### 1. Login to Pulumi Cloud and initialize stack
+
+Let's begin by logging into Pulumi Cloud
+
+```bash
+# There are many ways to store Pulumi state, here we use Pulumi Cloud
+# Other state backends include S3, local file, and more
+pulumi login
+```
+
+### 2. Prepare a new Pulumi Project
+
+Here we create our new pulumi project directory and populate it from the `sagemaker-aws-python` [Pulumi template](https://github.com/pulumi/templates)
+
+```bash
+# Create a new directory & change directories into it
+mkdir newSagemaker && cd newSagemaker
+
+# Start your project from the sagemaker-aws-python template
+# Follow along with the prompts to create your new project and initialize a stack
+pulumi new sagemaker-aws-python
+```
+
+### 3. Deploy your model as a new SageMaker endpoint
+
+This step may take between 10 and 20 minutes while Amazon AWS builds your infrastructure. You can follow along in the console as resources are provisioned, or open the link displayed in terminal for your stack in a browser to follow along there too!
+
+```bash
+pulumi up
+```
+
+## Try your new SageMaker Endpoint
+
+Once your stack has finished deploying, use this rudimentary Python snippet to test the deployed SageMaker endpoint.
+
+> NOTE: notice that we are using `us-east-1` in this script, be sure to change the reagion in python to match the reagion you deployed the SageMaker endpoint into.
+
+### 4. Save the following python snippet as `test.py`
+
+```python
+import json, boto3, argparse
+
+def main(endpoint_name):
+    client = boto3.client('sagemaker-runtime', region_name='us-east-1')
+    payload = json.dumps({"inputs": "In 3 words, name the biggest mountain on earth?"})
+    response = client.invoke_endpoint(EndpointName=endpoint_name, ContentType="application/json", Body=payload)
+    print("Response:", json.loads(response['Body'].read().decode()))
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("endpoint_name")
+    main(parser.parse_args().endpoint_name)
+```
+
+### 5. Run the test prompt
+
+Now source the local `venv` and run the script using the AWS name of your new endpoint from pulumi state outputs:
+
+```bash
+# On Linux & MacOS
+source venv/bin/activate
+
+# Execute test.py
+python3 test.py $(pulumi stack output EndpointName)
+```
+
+### 6. Cleanup / Teardown
+
+Finally, be sure to clean up un-used resources when you're done with one easy command.
+
+```bash
+pulumi destroy
+```
+
+## Conclusion
+
+To recap, in a few commands, we created a new Pulumi project from a ready to roll template, deployed an LLM Model endpoint on AWS SageMaker, and tested it with a short python script to generate a response from our model!
+
+AI and ML is becoming more of a necessity every day. It may appear daunting or out of reach at first glance, but with the power of IaC written in Pulumi Python programs, getting started has never been easier. 
```

Comment:
This is the kind of high-level justification I mentioned having earlier in the post. Maybe move this up there, or copy it up there and recap it here as a reminder?

Reply:
well this justification is still too nebulous. i would like to see the exact details and customer pain